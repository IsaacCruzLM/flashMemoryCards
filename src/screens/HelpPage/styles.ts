import {StyleSheet} from 'react-native';
import themes from '../../styles/themes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: themes.spacing.unit * 2.5,
  },
  title: {
    fontSize: themes.typography.fontSizeLargeTitle,
    fontWeight: 'bold',
    marginBottom: themes.spacing.unit * 2.5,
    color: themes.colors.primary,
  },
  section: {
    marginBottom: themes.spacing.unit * 2.5,
  },
  subtitle: {
    fontSize: themes.typography.fontSizeTitle,
    fontWeight: 'bold',
    marginBottom: themes.spacing.unit * 1.25,
    color: themes.colors.primary,
  },
  text: {
    fontSize: themes.typography.fontSizeText,
    lineHeight: themes.spacing.unit * 3,
    color: themes.colors.text,
  },
  sectionVersion: {
    flex: 1,
    justifyContent: 'flex-end',
  },
  subtitleVersion: {
    fontSize: themes.typography.fontSizeTitle,
    fontWeight: 'bold',
    color: themes.colors.primary,
    textAlign: 'center',
  },
  textVersion: {
    fontSize: themes.typography.fontSizeText,
    lineHeight: themes.spacing.unit * 3,
    color: themes.colors.text,
    textAlign: 'center',
  },
});

export default styles;
