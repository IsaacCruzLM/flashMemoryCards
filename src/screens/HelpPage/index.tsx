import React from 'react';
import {View, Text, ScrollView} from 'react-native';

import styles from './styles';

const HelpPage = () => {
  return (
    <ScrollView contentContainerStyle={styles.container}>
      <Text style={styles.title}>Como Utilizar Memory Cards</Text>

      <View style={styles.section}>
        <Text style={styles.subtitle}>1. Criar uma Anotação</Text>
        <Text style={styles.text}>
          Para criar uma nova anotação, basta clicar no ícone "+" na tela
          inicial ou na tela de uma categoria especifica e preencher os
          detalhes. Você pode adicionar um título e o conteúdo da sua anotação.
        </Text>
      </View>

      <View style={styles.section}>
        <Text style={styles.subtitle}>2. Revisar Anotações</Text>
        <Text style={styles.text}>
          O app enviará lembretes automáticos para você revisar suas anotações
          em 1 dia, 7 dias, 1 mês, e 6 meses. Você pode acessar esses lembretes
          diretamente pela notificação. Lembre-se de ativar as notificações!
        </Text>
      </View>

      <View style={styles.section}>
        <Text style={styles.subtitle}>3. Editar ou Excluir Anotações</Text>
        <Text style={styles.text}>
          Para editar ou excluir uma anotação, toque na anotação desejada e
          utilize os botões de edição ou exclusão no topo da tela.
        </Text>
      </View>

      <View style={styles.section}>
        <Text style={styles.subtitle}>4. Criar Categorias e Assuntos</Text>
        <Text style={styles.text}>
          Para organizar melhor suas anotações, você pode criar categorias e
          assuntos personalizados. Isso ajudará a separar suas anotações em
          grupos relacionados. Vá até o menu lateral, acesse a seção
          correspondente e adicione novas categorias e assuntos conforme suas
          necessidades.
        </Text>
      </View>

      <View style={styles.section}>
        <Text style={styles.subtitle}>5. Gerar Resumo PDF</Text>
        <Text style={styles.text}>
          Se desejar gerar um PDF a partir de suas anotações, vá até a opção
          "Criar Resumo PDF" no menu lateral e selecione as características das
          anotações que deseja incluir.
        </Text>
      </View>

      <View style={styles.section}>
        <Text style={styles.subtitle}>6. Enviar Feedback</Text>
        <Text style={styles.text}>
          Caso tenha encontrado algum problema ou tenha sugestões de melhorias,
          acesse o menu lateral e selecione "Feedback" para nos enviar sua
          opinião.
        </Text>
      </View>
      <View style={styles.sectionVersion}>
        <Text style={styles.subtitleVersion}>Versão 1.0.0</Text>
        <Text style={styles.textVersion}>Criado por: Isaac Cruz</Text>
      </View>
    </ScrollView>
  );
};

export default HelpPage;
