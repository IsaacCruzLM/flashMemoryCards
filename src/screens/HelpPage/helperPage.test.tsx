import React from 'react';
import renderer from 'react-test-renderer';
import HelpPage from './index';

describe('HelpPage', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<HelpPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle user interactions correctly', () => {
    const component = renderer.create(<HelpPage />);
    const instance = component.root;

    const helpTitle = instance.findByProps({testID: 'help-title'});
    expect(helpTitle).toBeTruthy();
  });

  it('should render all sections', () => {
    const component = renderer.create(<HelpPage />);

    const scrollContainer = component.root.findByType(
      'RCTScrollView' as React.ElementType,
    );

    const numberOfSections = scrollContainer.props.children.length;

    expect(numberOfSections).toBeGreaterThan(0);
  });
});
