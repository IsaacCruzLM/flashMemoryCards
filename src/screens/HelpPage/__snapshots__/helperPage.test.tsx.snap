// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpPage should render correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#22272E",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
      testID="help-title"
    >
      Como Utilizar Memory Cards
    </Text>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#22272E",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        1. Criar uma Anotação
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        Para criar uma nova anotação, basta clicar no ícone "+" na tela inicial ou na tela de uma categoria especifica e preencher os detalhes. Você pode adicionar um título e o conteúdo da sua anotação.
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#22272E",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        2. Revisar Anotações
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        O app enviará lembretes automáticos para você revisar suas anotações em 1 dia, 7 dias, 1 mês, e 6 meses. Você pode acessar esses lembretes diretamente pela notificação. Lembre-se de ativar as notificações!
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#22272E",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        3. Editar ou Excluir Anotações
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        Para editar ou excluir uma anotação, toque na anotação desejada e utilize os botões de edição ou exclusão no topo da tela.
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#22272E",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        4. Criar Categorias e Assuntos
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        Para organizar melhor suas anotações, você pode criar categorias e assuntos personalizados. Isso ajudará a separar suas anotações em grupos relacionados. Vá até o menu lateral, acesse a seção correspondente e adicione novas categorias e assuntos conforme suas necessidades.
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#22272E",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        5. Gerar Resumo PDF
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        Se desejar gerar um PDF a partir de suas anotações, vá até a opção "Criar Resumo PDF" no menu lateral e selecione as características das anotações que deseja incluir.
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#22272E",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        6. Enviar Feedback
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
      >
        Caso tenha encontrado algum problema ou tenha sugestões de melhorias, acesse o menu lateral e selecione "Feedback" para nos enviar sua opinião.
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#22272E",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Versão 1.0.0
      </Text>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": 14,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Criado por: Isaac Cruz
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
